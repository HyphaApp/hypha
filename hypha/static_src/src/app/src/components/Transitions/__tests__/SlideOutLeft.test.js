import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import SlideOutLeft from "../SlideOutLeft";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Slide out left component", () => {
        
    const children = <span>Children</span>
    const subject = mount(<SlideOutLeft>{children}</SlideOutLeft>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    test('render a Slide out left component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
