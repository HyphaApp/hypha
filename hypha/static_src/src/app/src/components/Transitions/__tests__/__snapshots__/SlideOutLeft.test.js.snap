// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Slide out left component render a Slide out left component 1`] = `
<SlideOutLeft>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={250}
    unmountOnExit={false}
  >
    <div
      style={
        Object {
          "position": "absolute",
          "transform": "translate3d(0, 0, 0)",
          "transition": "transform 250ms ease-in-out",
          "width": "100%",
        }
      }
    >
      <span>
        Children
      </span>
    </div>
  </Transition>
</SlideOutLeft>
`;
